### PR TITLE
chore: update permissions

### PR DESCRIPTION
**Pull request by sympatheticmoose**
**What this PR does / why we need it**:

Introduces top level read-only permissions in the CI workflow

Moves write permissions to within the single job (kind of a vanity change but stops certain security tooling from complaining)

**Which issue(s) this PR fixes**:

part of https://github.com/grafana/plugin-tools/issues/742

**Special notes for your reviewer**:

Tested with a custom plugin and CI/Release still appeared to execute as expected. 
CI workflow may require more changes as per the issue

### CODE REVIEW

**Review by leventebalogh — COMMENTED** [review]
Hmm, maybe I am doing something wrong, but this didn't work for me ([failing workflow](https://github.com/grafana/plugin-workflow-tester/actions/runs/7883330266/workflow)). Are you sure `read-only` is a valid permission value? As far as I can see [here](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#defining-access-for-the-github_token-scopes) to define access for all the available scopes we can either use `read-all` or `write-all`.

**Comment by leventebalogh** [comment]
(Maybe it would be useful to automatically open a PR in the https://github.com/grafana/plugin-workflow-tester/ repo if any of the workflow template files change to see if they pass. However as we don't edit these too often maybe it doesn't worth the effort.)

**Comment by sunker** [comment]
> (Maybe it would be useful to automatically open a PR in the https://github.com/grafana/plugin-workflow-tester/ repo if any of the workflow template files change to see if they pass. However as we don't edit these too often maybe it doesn't worth the effort.)

Nice, did not know we had this repo.

**Comment by sympatheticmoose** [comment]
> Hmm, maybe I am doing something wrong, but this didn't work for me ([failing workflow](https://github.com/grafana/plugin-workflow-tester/actions/runs/7883330266/workflow)). Are you sure `read-only` is a valid permission value? As far as I can see [here](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#defining-access-for-the-github_token-scopes) to define access for all the available scopes we can either use `read-all` or `write-all`.

🤦

File: packages/create-plugin/templates/github/ci/.github/workflows/release.yml [hunk 1]
```diff
@@ -8,11 +8,12 @@ on:
     tags:
       - 'v*' # Run workflow on version tags, e.g. v1.0.0.
 
-permissions:
-  contents: write
+permissions: read-all
```
Comment (leventebalogh) [inline packages/create-plugin/templates/github/ci/.github/workflows/release.yml:11]: As we only have a single job in this workflow, what's the reason for this change at this point?

**Review by leventebalogh — APPROVED** [review]
